### PR TITLE
feat: store and display template owner

### LINKED ISSUE #2151: Templates view and template view should identify template owner
In the very near future template owners will be a minority of users in a Coder installation (already true in ours) and a minority of users of any particular template.  Both the view listing the templates and the template detail view should identify that owner.

### PR DESCRIPTION
**Pull request by AbhineetJain**
This PR stores the template owner in the database and displays that information in the UI.

## Subtasks

- [x] add migrations to store owner_id in Templates table.
- [x] update the API calls to store owner_id and return owner_id and owner_name
- [x] display owner_name on Templates page
- [x] display owner_name on Template page
- [x] add new story
- [x] update unit tests

Fixes #2151 

### Templates page
Screen Shot 2022-06-09 at 11 07 33 PM

### Template page
Screen Shot 2022-06-09 at 11 06 46 PM

### CODE REVIEW

File: site/src/components/TemplateStats/TemplateStats.tsx [hunk 1]
```diff
@@ -45,6 +47,11 @@ export const TemplateStats: FC<TemplateStatsProps> = ({ template, activeVersion
           {dayjs().to(dayjs(template.updated_at))}
         </span>
       </div>
+      <div className={styles.statsDivider} />
+      <div className={styles.statItem}>
```
Comment (Kira-Pilot) [inline site/src/components/TemplateStats/TemplateStats.tsx:51]: Just an idea: what do you think about changing the `min-width` on the `statItem` class to be `20%`? That way, header items would expand to fill the row a bit better, i.e:
Screen Shot 2022-06-10 at 10 41 39 AM

File: site/src/testHelpers/entities.ts [hunk 2]
```diff
@@ -114,6 +114,8 @@ export const MockTemplate: TypesGen.Template = {
   description: "This is a test description.",
   max_ttl_ms: 604800000,
   min_autostart_interval_ms: 3600000,
+  owner_id: "test-owner-id",
+  owner_name: "test_owner",
```
Comment (Kira-Pilot) [inline site/src/testHelpers/entities.ts:118]: 🥳

**Review by Kira-Pilot — APPROVED** [review]
FE looks good! ✔️

File: coderd/templates.go [hunk 3]
```diff
@@ -400,29 +406,36 @@ func (api *API) patchTemplateMeta(rw http.ResponseWriter, r *http.Request) {
...
 	}
 	return apiTemplates
 }
 
-func convertTemplate(template database.Template, workspaceOwnerCount uint32) codersdk.Template {
+func convertTemplate(ctx context.Context, db database.Store, template database.Template, workspaceOwnerCount uint32) codersdk.Template {
+	var ownerName string
+	if template.OwnerID.Valid {
+		owner, err := db.GetUserByID(ctx, template.OwnerID.UUID)
+		if err == nil {
```
Comment (mafredri) [inline coderd/templates.go:435]: We should ideally propagate context cancellation here somehow.

This is just an example scenario:

We receive a request, we decide it should at most take 1 second:

```go
ctx, _ := context.WithTimeout(r.Context(), 1 * time.Second)
```

(The error could also be from the DB, connection issue, statement timeout, etc.)

This context would then be passed on to `convertTemplate`(s) and as a result, all of the `ownerName`s would likely become the empty string. In this instance it's not the end of the world, but it's a hidden silent failure which isn't great. IMO we should strive to always fail correctly.

**Review by mafredri — COMMENTED** [review]
Looks good! There's just one gotcha that I think would be worth fixing, unfortunately fixing it requires some refactoring.

File: site/src/components/TemplateStats/TemplateStats.tsx (hunk 1 above)
Comment (AbhineetJain) [reply site/src/components/TemplateStats/TemplateStats.tsx:51]: Sounds like a great idea! Also, a step towards the design being more responsive. 🎉

**Review by Emyrk — CHANGES_REQUESTED** [review]
Wait I'm confused, why does a template have a user as an owner?

Is it really an "owner" or just a "created_by" for audit log purposes?

**Comment by AbhineetJain** [comment]
> Wait I'm confused, why does a template have a user as an owner?
> 
> Is it really an "owner" or just a "created_by" for audit log purposes?

As per https://github.com/coder/coder/issues/2151, we want to identify the owner of the templates in the UI on account of them being minority users in an eventually grown Coder installation.

File: coderd/templates.go (hunk 3 above)
Comment (AbhineetJain) [reply coderd/templates.go]: Handled errors in the caller methods as discussed.

**Review by Emyrk — APPROVED** [review]

**Pull request merged**